engine/graphics/image.cpp

```diff
@@ -0,0 +1,56 @@
+#include "image.h"
+
+#include "utils.h"
+
+namespace pong
```

Comment:
I believe since 17 you can nest namespaces more naturally.

This would become 

```
namespace pong::image 
{
...
} 
```

Reply:
So along with introduced the Image class I put the `pong::image` namespace with those functions, but the `Image` class under just `pong`  I think it is a bit nicer to be able to create an `Image` without doing something like `image::Image` but but at that point maybe those functions should just be static functions like before.  That would at least make more sense than being under the `Texture` class previously. Any thoughts?

Reply:
> think it is a bit nicer to be able to create an Image without doing something like image::Image but but at that point maybe those functions should just be static functions like before.

Authors choice. I have seen many times something like `pong::image::Image`. Often that's where you get things like `using pong::image::Image` which allows you to just use it as `Image` in the file where that is called, kind of the best of both worlds.

Reply:
Yeah thats a good point. Kind of forgot about `using` making that not an issue haha